server/src/routes/userRoutes.ts

```diff
@@ -1,56 +1,54 @@
+import { AppDataSource } from './../database'; // Import your TypeORM DataSource
+import { User } from './../entities/User'; // Import the User entity
 import bcrypt from 'bcryptjs';
 
 export const router = express.Router();
 
-router.get('/', (req, res) => {
-    pool.query('SELECT * FROM users', (err, results) => {
-      if (err) {
-        console.error('Error fetching users:', err);
-        return res.status(500).send('Failed to fetch users');
-      }
-  
-      res.status(200).json(results);
-    });
-  });
-
-// USER AUTHENTICATION
-// Register Route Handler
+// Get all users (using TypeORM repository)
+router.get('/', async (req: Request, res: Response) => {
+  try {
+    const userRepository = AppDataSource.getRepository(User);
+    const users = await userRepository.find(); // Fetch all users
+    res.status(200).json(users);
+  } catch (err) {
+    console.error('Error fetching users:', err);
+    res.status(500).send('Failed to fetch users');
+  }
+});
+
+// Register Handler (user registration)
 const registerHandler: RequestHandler = async (req: Request, res: Response): Promise<void> => {
-    const { username, password } = req.body;
-  
-    if (!username || !password) {
-      res.status(400).json({ message: 'Username and password are required' });
+  const { username, password } = req.body;
+
+  if (!username || !password) {
+    res.status(400).json({ message: 'Username and password are required' });
+    return;
+  }
+
+  try {
+    const userRepository = AppDataSource.getRepository(User);
+
+    // Check if the username is already taken
+    const existingUser = await userRepository.findOneBy({ username });
+    if (existingUser) {
+      res.status(400).json({ message: 'Username already taken' });
       return;
     }
-  
-    try {
-      // Check if user already exists
-      const [results]: [mysql.RowDataPacket[], mysql.FieldPacket[]] = await pool.promise().query('SELECT * FROM users WHERE username = ?', [username]);
-  
-      if (results.length > 0) {
-        res.status(400).json({ message: 'Username already taken' });
-        return;
-      }
-  
-      // Hash the password
-      const hashedPassword = await bcrypt.hash(password, 10);
-  
-      // Insert new user into database
-      const [result]: [mysql.ResultSetHeader, mysql.FieldPacket[]] = await pool.promise().query('INSERT INTO users (username, password) VALUES (?, ?)', [username, hashedPassword]);
-  
-      // Checking if the insertion was successful by accessing the `affectedRows` property
-      if (result.affectedRows > 0) {
-        res.status(201).json({ message: 'User registered successfully' });
-      } else {
-        res.status(500).json({ message: 'Error registering user' });
-      }
-    } catch (err) {
-      console.error(err);
-      res.status(500).json({ message: 'Database or server error' });
-    }
+
+    // Hash the password
+    const hashedPassword = await bcrypt.hash(password, 10);
```

Comment:
Ok I see you did it here, whatever you prefer

Reply:
You have other suggestions? Shay worked on it

Reply:
They are both good ( the comment I added on the DB entity is more strict). Like it is always make sure that the password is hashed ( In the DB level ). BUT for us it is fine, if you checked it with this code lets stay with it.